h/security/policy/_cookie.py

```diff
@@ -41,7 +51,9 @@ def remember(self, request, userid, **kw):  # pylint:disable=unused-argument
             request.session.update(data)
             request.session.new_csrf_token()
 
-        return request.find_service(AuthCookieService).create_cookie(userid)
+        ticket_id = base64.urlsafe_b64encode(urandom(32)).rstrip(b"=").decode("ascii")
```

Comment:
This could be `generate_ticket` or similar in `AuthTicketService` if we don't think different identities that uses tickets care that much about the ticket ID structure.

Reply:
Yeah, maybe. The ID generation is outside of the ticket service in `pyramid-authsanity`'s design (and in h's previous authsanity-inspired design). The one reason I can think of for it is in case different users of `AuthTicketService` had different requirements for the keys (e.g. because they need to be serializable in a certain way, or some other reason). But if nothing else the key generation could be a `@staticmethod` on `AuthTicketService` itself. I'll send a follow-up PR to do that: https://github.com/hypothesis/h/pull/8874